functions.php

```diff
@@ -617,3 +617,52 @@ function ($theme_json) {
         return $theme_json->update_with($new_color_palette);
     }
 );
+
+// Registering into the Settings > Writing setting page
+add_action(
+    'admin_menu',
+    function (): void {
+        register_setting('writing', 'revisions_to_keep');
+
+        add_settings_field(
+            'post-revisions-field',
+            __('Post revisions', 'planet4-master-theme-backend'),
+            function ($val): void {
+                $id = $val['id'];
+                $option_name = $val['option_name'];
+                ?>
+                    <input
+                        type="number"
+                        name="<?php echo esc_attr($option_name) ?>"
+                        id="<?php echo esc_attr($id) ?>"
+                        value="<?php echo esc_attr(get_option($option_name)) ?>"
+                    />
+                    <span>
+                        <?php echo __(
+                            'Maximum number of revisions to store for each post.',
+                            'planet4-master-theme-backend'
+                        )?>
+                    </span>
+                <?php
+            },
+            'writing',
+            'default',
+            array(
+                'id' => 'post-revisions-field',
+                'option_name' => 'revisions_to_keep'
+            )
+        );
+    },
+    10,
+    2,
+);
+
+// Maximum revisions to keep whenever the editor save a post
+add_filter(
```

Comment:
We can make this function a bit more safe, -(In case of `revisions_to_keep` found empty...)
```
add_filter(
    'wp_revisions_to_keep',
    function ($num) {
        $revisions_to_keep = get_option('revisions_to_keep');
        return $revisions_to_keep ?: $num;
    },
    10,
    1
);
```

Reply:
We can move this add_filter hook to  `src/MasterSite.php` file as well.
Note: The reason we move it out of `functions.php` is, this file will grow very fast if we add filters and hooks in it.